File: src/main/java/com/nao4j/subtrol/document/internal/Quantity.java
package com.nao4j.subtrol.document.internal;

import lombok.Data;
import lombok.NonNull;

@Data
public class Quantity {

    public enum QuantityType {
        UNITS,
        HOURS,
        DAYS,
        MONTHS,
        YEARS
    }

    private final long count;
    @NonNull
    private final QuantityType type;

    public Quantity(final long count, final QuantityType type) {
        this.count = count;
        this.type = type;

        if (count <= 0) {
            throw new IllegalArgumentException();
        }
    }

}

File: src/main/kotlin/com/nao4j/subtrol/model/Quantity.kt
package com.nao4j.subtrol.model

data class Quantity(val count: Long, val type: QuantityType) {

    enum class QuantityType {
        UNITS,
        HOURS,
        DAYS,
        MONTHS,
        YEARS
    }

    init {
        if (count <= 0) {
            throw IllegalArgumentException()
        }
    }

}
